pkg/controller/podautoscaler/metrics/client_test.go

```diff
@@ -429,7 +429,6 @@ func TestRESTClientContainerCPUEmptyMetricsForOnePod(t *testing.T) {
 		container:          "test-1",
 		targetTimestamp:    targetTimestamp,
 		window:             window,
-		desiredError:       fmt.Errorf("failed to get container metrics"),
```

Comment:
Can we add a test for the case where we are returning metrics for a container that is missing?

Reply:
So this is that test, we just don't expect it to error now. I could change the case here to look for the container that is actually missing in the metrics and check that it isn't returned if that's what we're looking for

Reply:
If anything, i think this test actually highlights the underlying bug here. The container in the test case we're asking for metrics on is `test-1` and both of the pods here have metrics for that container; its just one of them that is missing metrics for the second container and producing the error; for the container we didn't even ask for metrics on